File: thread_spawner.cpp
#include <thread>
#include <iostream>

void hello(int tid) {
	std::cout << "Hello from thread " << tid << std::endl;
}

int main(int argc, char *args[]) {
	std::thread threads[4];

	for (int i = 0; i < 4; i++) {
		threads[i] = std::thread(hello, i);
		threads[i].join();
	}
}

File: ThreadSpawner.java

public class ThreadSpawner extends Thread{

	private int threadID;

	public ThreadSpawner(int id) {
		threadID = id;
	}

	//when start() is called on each thread, will print Hello message.
	public void run() {
		System.out.println("Hello from thread " + threadID);
	}

	public static void main(String[] args) throws InterruptedException {
		//create array of the ThreadSpawner class
		ThreadSpawner[] threads = new ThreadSpawner[4];
		//for each element, create the thread, start it, and join it. 
		for (int i = 0; i < 4; ++i) {
			threads[i] = new ThreadSpawner(i);
			threads[i].start();
			threads[i].join();
		}
	}
}
